jerry-core/ecma/builtin-objects/ecma-builtin-array.c

```diff
@@ -283,41 +283,44 @@ ecma_builtin_array_object_from (ecma_value_t this_arg, /**< 'this' argument */
     goto cleanup;
   }
 
-  len_value = ecma_make_length_value (len);
-
   /* 12. */
-  ecma_value_t array;
+  ecma_object_t *array_obj_p;
 
   /* 12.a */
   if (ecma_is_constructor (constructor))
   {
     ecma_object_t *constructor_obj_p = ecma_get_object_from_value (constructor);
 
-    array = ecma_op_function_construct (constructor_obj_p, constructor_obj_p, &len_value, 1);
+    len_value = ecma_make_length_value (len);
+    ecma_value_t array = ecma_op_function_construct (constructor_obj_p, constructor_obj_p, &len_value, 1);
+    ecma_free_value (len_value);
```

Comment:
If we do a `len_value` free here we don't need the one in the `if` below.

Reply:
Good catch.